iOSMoviesApp/CustomTableViewCells/InfoTableViewCell/InfoTableViewCell.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  InfoTableViewCell.swift
+//  iOSMoviesApp
+//
+//  Created by Yash Agrawal on 16/07/25.
+//
+
+import UIKit
+
+protocol InfoTableViewCellDelegate: AnyObject {
+    func updatePoster(with imageData : Data, cacheKey : NSString)
+    func updatePosterFromCache(with image: UIImage)
+}
+
+class InfoTableViewCell: UITableViewCell {
+    
+    @IBOutlet weak var movieName: UILabel!
+    @IBOutlet weak var movieVotes: UILabel!
+    @IBOutlet weak var movieRating: UILabel!
+    @IBOutlet weak var movieDescription: UILabel!
+    @IBOutlet weak var movieGenre: UILabel!
+    @IBOutlet weak var MoviePoster: UIImageView!
+    
+    private var infoVM: InfoViewModel?
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        self.selectionStyle = .none
+    }
+    
+    func configureState(with infoVM: InfoViewModel?) {
+        guard let infoVM: InfoViewModel = infoVM else { return }
+        self.infoVM = infoVM
+        
+        setupMoviePoster()
+        setupMovieName()
+        setupMovieVotes()
+        setupMovieDescription()
+        setupMovieGenre()
+        setupMovieRating()
+    }
+    
+    func setupMoviePoster() {
```

Comment:
```suggestion
    private func setupMoviePoster() {
```

Make the necessary methods `private` throughout the class.